analyses/rnaseq-batch-correct/run_analysis.sh

```diff
@@ -0,0 +1,83 @@
+# GTEx + TARGET NBL (correct by study_id and library_type)
+# Rscript code/01-batch-correct.R \
+# --mat 'input/gtex-gene-expression-rsem-tpm-collapsed.polya.rds, input/target-nbl-gene-expression-rsem-tpm-collapsed.polya.rds' \
+# --type 'tpm' \
+# --metadata 'input/gtex-histologies.tsv, input/target-nbl-histologies.tsv' \
+# --id_col 'sample_id' \
+# --batch_cols 'study_id, library_type' \
+# --other_cols_to_keep 'group' \
+# --output_prefix 'gtex-target-nbl-gene-expression-rsem-tpm-collapsed.polya'
+
```

Comment:
Should these GTEx + TARGET NBL be uncommented ? Can't test the run_analysis.sh since it errors out saying  the output/gtex-target-gene-expression-rsem-tpm-collapsed* files are missing. Ideally I think this code should be run within this script ( but I can imagine batch correction over all gtex and target taking a lot of time so was commented out here? ). 

Do we want to implement the batch correction step as part of preprocessing so that we can download these files directly to `data` folder from s3 and plot UMAP in this module?  cc @jharenza for input

Reply:
Code was commented by mistake - I have uncommented it. I think all the tsv and rds files are on s3 except for the target NBL data because I had to subset that locally in order to test. Wasn't sure if this is meant to be uploaded to the bucket. Another thing is I had to create the KF NBL tsv locally to run this module - until it is standardized (the one on s3 does not have library prep info etc).

Reply:
Also, just so you are aware and in the interest of your time,  I am testing another batch correction method. There was some discussion around it in the OT meeting which @aadamk mentioned [here](https://github.com/PediatricOpenTargets/ticket-tracker/issues/10#issuecomment-844091380). You can treat this PR as draft for now, I suppose.

Reply:
OK sounds good, thanks!

Reply:
>Another thing is I had to create the KF NBL tsv locally to run this module - until it is standardized (the one on s3 does not have library prep info etc).

The `kfnbl-histologies.tsv` has `RNA_library` which has this information. Actually, and sorry to be a pain, but we should change this
```
# --batch_cols 'study_id, library_type' \<br class="Apple-interchange-newline">
 ```
to 
```
# --batch_cols 'study_id, RNA_library' \<br class="Apple-interchange-newline">
```

to match what is in `pbta-histologies.tsv`

Reply:
@komalsrathi I will update the target, tcga, gtex histologies files with that column name and add to our v2 folder so we can get the release out

Reply:
@komalsrathi the files are updated

Reply:
Ok sure, I will download those and let you know in a bit. 

I am also generating the counts (gtex, target and tcga) files - will let you know once those are done but would you want those uploaded for this or next release? (they are still working on my write access)

Reply:
Another issue is that we need consistent naming for sample identifiers and groups that are picked from histology files - right now picking `sample_id` and `group` from gtex, target and tcga and for KF NBL changing `Kids_First_Biospecimen_ID` to `sample_id` and `broad_histology` to `group`.  Do you have any ideas?

Reply:
oh yes - @ewafula is working on creating the TARGET file in this way (renaming `sample_id` to `Kids_First_Biospecimen_ID`) and will also work on TCGA. Instead of `group`, you can use `cohort`, which we have in `pbta-histologies.tsv`. I think we should probably make a bigger ticket about this.

Reply:
Actually I may have an idea to avoid all the local formatting but have to think it through. I'll update you

Reply:
Ok for testing purpose, I will also make those changes locally and test the module i.e. add RNA_library (library type), cohort (i.e. study) and Kids_First_Biospecimen_ID (i.e. sample_id).

`Group` actually does not refer to the cohort but either the normal tissue type for e.g. Liver or disease type for e.g. Neuroblastoma. For GTEx, we will have to rename the tissue type to either `broad_histology` or some other column from the histologies file that we will be using to color the data points in the plot.